test/framework/src/main/java/org/elasticsearch/index/mapper/BlockLoaderTestCase.java

```diff
@@ -0,0 +1,166 @@
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.tests.index.RandomIndexWriter;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.fieldvisitor.StoredFieldLoader;
+import org.elasticsearch.logsdb.datageneration.DataGeneratorSpecification;
+import org.elasticsearch.logsdb.datageneration.FieldDataGenerator;
+import org.elasticsearch.logsdb.datageneration.FieldType;
+import org.elasticsearch.logsdb.datageneration.MappingGenerator;
+import org.elasticsearch.logsdb.datageneration.Template;
+import org.elasticsearch.logsdb.datageneration.datasource.DataSourceHandler;
+import org.elasticsearch.logsdb.datageneration.datasource.DataSourceRequest;
+import org.elasticsearch.logsdb.datageneration.datasource.DataSourceResponse;
+import org.elasticsearch.search.fetch.StoredFieldsSpec;
+import org.elasticsearch.search.lookup.SearchLookup;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public abstract class BlockLoaderTestCase extends MapperServiceTestCase {
+    private final String fieldName;
+    private final Template template;
+    private final MappingGenerator mappingGenerator;
+    private final FieldDataGenerator generator;
+
+    protected BlockLoaderTestCase(FieldType fieldType) {
+        this.fieldName = randomAlphaOfLengthBetween(5, 10);
+
+        // Disable all dynamic mapping
+        var specification = DataGeneratorSpecification.builder()
+            .withFullyDynamicMapping(false)
+            .withDataSourceHandlers(List.of(new DataSourceHandler() {
+                @Override
+                public DataSourceResponse.DynamicMappingGenerator handle(DataSourceRequest.DynamicMappingGenerator request) {
+                    return new DataSourceResponse.DynamicMappingGenerator(isObject -> false);
+                }
+            }))
+            .build();
+
+        this.template = new Template(Map.of(fieldName, new Template.Leaf(fieldName, fieldType)));
+        this.mappingGenerator = new MappingGenerator(specification);
+        this.generator = fieldType.generator(fieldName, specification.dataSource());
+    }
+
+    public void testBlockLoader() throws IOException {
+        var mapping = mappingGenerator.generate(template);
+        var mappingXContent = XContentBuilder.builder(XContentType.JSON.xContent()).map(mapping.raw());
+
+        var syntheticSource = randomBoolean();
```

Comment:
Seeing this makes me wonder how much we should parameterize vs randomize the mappings. It's more work to make a parameterized array, but these feel like something a couple of nested `for` loops could spit out the valid cases and we could see them. It there aren't thousands of them we could run them on every execution.

No need to change anything in this PR. I'd be fine flipping later if you think it's a good idea. Or never. It's probably fine.

Reply:
Yes, this is a good question. My statement here is definitely more "i need mappings and values and there is already this thing that knows how to do that" than "randomize is a way to go". I'll keep it but i can definitely see some evolution.